keras/src/models/model.py

```diff
@@ -556,6 +556,112 @@ def _get_variable_map(self):
         map_saveable_variables(self, store=store, visited_saveables=set())
         return store
 
+    def get_state_tree(self):
+        """Retrieves tree-like structure of model variables.
+
+        This method allows retrieval of different model variables (trainable,
+        non-trainable, optimizer, and metrics). The variables are returned in a
+        nested dictionary format, where the keys correspond to the variable
+        names and the values are the nested representations of the variables.
+
+        Returns:
+            dict: A dictionary containing the nested representations of the
+            requested variables. The keys are the variable names, and the values
+            are the corresponding nested dictionaries.
```

Comment:
Please add a code example for a simple compiled model (e.g. a Sequential via 1 layer in it)

Something like

```python
model = keras.Sequential([
    keras.Input(shape=(32,), name="my_input"),
    keras.layers.Dense(1, activation="sigmoid", name="my_dense"),
], name="my_sequential")
model.compile(optimizer="adam", loss="mse", metrics=["mae"])
model.fit(...)
state_tree = model.get_state_tree()
```

Returns:

```
...
```

Reply:
Added an example! Let me know if it looks good.